src/op_mode/vrrp.py

```diff
@@ -13,47 +13,308 @@
+        instance['data']['wantstate'] = VRRP_STATE_TO_NAME.get(instance['data'].get('wantstate'), 'unknown')
+        instance['data']['auth_type'] = VRRP_AUTH_TO_NAME.get(instance['data'].get('auth_type'), 'unknown')
+        _process_field(instance['data'], 'lower_prio_no_advert', 'false', 'true')
+        _process_field(instance['data'], 'higher_prio_send_advert', 'true', 'false')
+        _process_field(instance['data'], 'accept', 'Enabled', 'Disabled')
+        _process_field(instance['data'], 'notify_deleted', 'Deleted', 'Fault')
+        _process_field(instance['data'], 'smtp_alert', 'yes', 'no')
+        _process_field(instance['data'], 'nopreempt', 'Disabled', 'Enabled')
+        _process_field(instance['data'], 'promote_secondaries', 'Enabled', 'Disabled')
+        instance['data']['vips'] = instance['data'].get('vips', False)
+        instance['data']['evips'] = instance['data'].get('evips', False)
+        instance['data']['vroutes'] = instance['data'].get('vroutes', False)
+        instance['data']['vrules'] = instance['data'].get('vrules', False)
+
+        instances.append(instance['data'])
+
+    return detail_template.render(instances=instances)
+
+
+def show_detail(raw: bool, group_name: typing.Optional[str] = None) -> typing.Union[list, str]:
+    """
+    Display detailed information about the VRRP group.
+
+    Args:
+        raw (bool): If True, return raw data instead of formatted output.
+        group_name (str, optional): Filter the data by a specific group name,
+            if provided.
+
+    Returns:
+        list or str: Raw data if `raw` is True, otherwise a formatted detail
+            output.
+    """
+    data = _get_raw_data(group_name)
+
+    if raw:
+        return data
+
+    return _get_formatted_detail_output(data)
+
+
+def show_statistics(raw: bool, group_name: typing.Optional[str] = None) -> typing.Union[list, str]:
+    """
+    Display VRRP group statistics.
+
+    Args:
+        raw (bool): If True, return raw data instead of formatted output.
+        group_name (str, optional): Filter the data by a specific group name,
+            if provided.
+
+    Returns:
+        list or str: Raw data if `raw` is True, otherwise a formatted statistic
+            output.
+    """
+    data = _get_raw_data(group_name)
+
+    if raw:
+        return data
+
+    return _get_formatted_statistics_output(data)
+
+
+def show_summary(raw: bool) -> typing.Union[list, str]:
+    """
+    Display a summary of VRRP group.
+
+    Args:
+        raw (bool): If True, return raw data instead of formatted output.
+
+    Returns:
+        list or str: Raw data if `raw` is True, otherwise a formatted summary output.
+    """
+    data = _get_raw_data()
+
+    if raw:
+        return data
+
+    return VRRP.format(data)
+
+
+if __name__ == '__main__':
+    try:
+        res = vyos.opmode.run(sys.modules[__name__])
+        if res:
+            print(res)
+    except (ValueError, vyos.opmode.Error, VRRPNoData) as e:
```

Comment:
As I mentioned above, we shouldn't allow raw `VRRPNoData` to bubble up this far. It should be re-raised as `DataUnavailable`, so that all op mode (CLI and API) can handle it uniformly.

Reply:
fixed